arena-pot/src/main/scala/com/uqbar/aop/transaction/ObjectTransactionImpl.scala

```diff
@@ -206,10 +203,8 @@ class ObjectTransactionImpl(var parent: ObjectTransactionImpl, var owner: TaskOw
             try {
               ReflectionUtils.invokeSetter(keyWrapper.getKey(), k, v);
             } catch {
-              case e: RuntimeException => throw new ProgramException(e) //
-                .addInfo("Object", keyWrapper.getKey()) //
-                .addInfo("Field", k) //
-                .addInfo("Value", v);
+              case e: RuntimeException =>
```

Comment:
jaja.. es @fdodino.. siempre es @fdodino

2016-08-11 10:34 GMT-03:00 Nico Passerini <notifications@github.com>:

> In arena-pot/src/main/scala/com/uqbar/aop/transaction/
> ObjectTransactionImpl.scala
> <https://github.com/uqbar-project/arena/pull/8#discussion_r74422309>:
>
> > @@ -206,10 +203,8 @@ class ObjectTransactionImpl(var parent: ObjectTransactionImpl, var owner: TaskOw
> >              try {
> >                ReflectionUtils.invokeSetter(keyWrapper.getKey(), k, v);
> >              } catch {
> > -              case e: RuntimeException => throw new ProgramException(e) //
> > -                .addInfo("Object", keyWrapper.getKey()) //
> > -                .addInfo("Field", k) //
> > -                .addInfo("Value", v);
> > +              case e: RuntimeException =>
>
> En el caso de una program exception, ni siquiera se cambiaba el mensaje,
> sólo se le agrega mediante addInfo la información de contexto. Javi, por
> qué estás revolviendo esto? No querés que te busque problemas más actuales
> para revisar? 2016-08-11 14:14 GMT+02:00 javierfernandes <
> notifications@github.com>:
> … <#m_69318127413193279_>
> In arena-pot/src/main/scala/com/uqbar/aop/transaction/
> ObjectTransactionImpl.scala <#8 (comment)
> <https://github.com/uqbar-project/arena/pull/8#discussion_r74410765>>: >
> @@ -206,10 +203,8 @@ class ObjectTransactionImpl(var parent:
> ObjectTransactionImpl, var owner: TaskOw > try { >
> ReflectionUtils.invokeSetter(keyWrapper.getKey(), k, v); > } catch { > -
> case e: RuntimeException => throw new ProgramException(e) // > -
> .addInfo("Object", keyWrapper.getKey()) // > - .addInfo("Field", k) // > -
> .addInfo("Value", v); > + case e: RuntimeException => No veo ningún smell
> en que una excepción sea mutable. Los objetos en general son mutables y se
> cumplen con eso de que "se comparten entre objetos y contextos". La
> ProgramException, de nuevo, era una exception pensada para ser lanzada por
> un framework, y no debería ser wrappeada por una excepción de negocio. O
> sea si se quedó sin espacio en disco, o no se pudo conectar a tal servidor,
> no podés wrappearla a un problema de dominio, es un problema puramente
> técnico. Entonces, por un lado, en forma medio manual, teníamos controles
> sobre esto. No se podía wrappear una program con una business. Por otro
> lado, esto tenía el efecto de que tal vez algunos problemas explotaban "muy
> abajo" y uno no entendía el contexto de la excepción. Entonces los addInfo,
> eran una manera de agregarle información al volver del stack. También, eran
> una forma de evitar el patrón de catcheo para wrappear, sólo para agregar
> un dato a mi exception. Eso es un patrón que hacemos seguido, y que no se
> si tiene mucho sentido. Es medio costumbre catch (RuntimeException e) {
> throw new RuntimeException("No pude persistir el objeto", e) } Necesito una
> exception nueva para eso ? simplemente para agregar un mensaje ? — You are
> receiving this because you were mentioned. Reply to this email directly,
> view it on GitHub <https://github.com/uqbar-project/arena/pull/8/files/
> 797af34680cac9217346bc335da039d79ab65424#r74410765>, or mute the thread <
> https://github.com/notifications/unsubscribe-auth/AEa1OcuQTd1NiZBdBpdWg-
> jvPKjkw5Swks5qexJDgaJpZM4C5ng4> .
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/uqbar-project/arena/pull/8/files/797af34680cac9217346bc335da039d79ab65424#r74422309>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEORWHmSxNOSS7tgBj8XGiAxj7Rryuseks5qeyT3gaJpZM4C5ng4>
> .
>